File: clients/web/src/lib/models/room.ts
import type { RoomDto } from '$lib/api/resources/rooms';
import type { UUID } from 'crypto';
import type { Profile } from './profile';

export interface Room {
	id: UUID;
	name: string;
	workspaceId: UUID;
	color: string;
	participants: Profile[];
}

export function createRoom(dto: RoomDto): Room {
	return {
		id: dto.id! as UUID,
		name: dto.name!,
		workspaceId: dto.workspaceId! as UUID,
		color: '000000',
		participants: dto.participants! as Profile[]
	};
}

File: backend/src/main/java/com/andhug/relay/room/api/Room.java
package com.andhug.relay.room.api;

import com.andhug.relay.profile.Profile;
import lombok.Getter;
import lombok.Setter;

import java.util.List;
import java.util.UUID;

@Getter
@Setter
public class Room {

    private UUID id;

    private String name;

    private UUID workspaceId;

    private RoomType type;

    private List<Profile> participants;
}
